packages/utils/src/utils/bytecode.ts

```diff
@@ -3,22 +3,28 @@ import { gzipSync, gunzipSync } from 'fflate';
 
 import { arrayify } from './arrayify';
 
-export const compressBytecode = (bytecode?: BytesLike) => {
-  if (!bytecode) {
+export const compressBytecode = (bytecodeAsBinary?: BytesLike) => {
+  if (!bytecodeAsBinary) {
     return '';
   }
 
-  const bytecodeBytes = arrayify(bytecode);
-  const bytecodeGzipped = gzipSync(bytecodeBytes);
-  const bytecodeEncoded = Buffer.from(bytecodeGzipped).toString('base64');
+  const bytecodeCompressBytes = arrayify(bytecodeAsBinary);
+  const bytecodeCompressGzipped = gzipSync(bytecodeCompressBytes);
+  const bytecodeCompressBinary = String.fromCharCode.apply(
```

Comment:
Just a suggestion to reduce the type casting

```suggestion
const bytecodeCompressBinary = Uint8Array.from(bytecodeCompressGzipped).reduce(
    (str, byte) => str + String.fromCharCode(byte),
    ''
  );
```

Reply:
`apply` is significantly more performant than `reduce` here so will resolve.